ooasp/run.py

```diff
@@ -231,32 +278,30 @@ def on_model(m):
     step_size = 1
     stats = {}
     done = False
+
     while not done:
-        print(f"\nSolving for size {next_id-1}...")
+        added = create_from_cautious(ctl, next_id, project=args.project)
+        next_id += added
+        if added > 0:
+            continue
+        # save_png("\n".join([str(c)+"." for c in config]), directory="out/solve")
+        log(f"\n==============================")
+        log(f"Solving for size {next_id-1}...")
+        ctl.configuration.solve.models = "1"
         res = ctl.solve(on_model=on_model)
         if res.satisfiable:
-            print("     Found model")
+            log("     Found model")
             done = True
-            break
-        print(f"    No configuration found for size {next_id-1}")
-        start_size = next_id
-        stats[next_id - 1] = ctl.statistics["summary"]["times"]
-        try_new = True
-        if use_cautious_generate:
-            added = create_from_cautious(ctl, next_id, project=args.project)
-            next_id += added
-            try_new = added == 0
-        while try_new:
-            ground(ctl, next_id, "object")
-            next_id += 1
-            if next_id >= start_size + step_size:
-                try_new = False
+            continue
+        log(f"No solution found")
+        ground(ctl, next_id, "object")
+        next_id += 1
 
     end = time.time()
     stats[next_id - 1] = ctl.statistics["summary"]["times"]
 
-    print("Done!")
-    out_name = f"benchmarks/latest/{generate_output_path(args)}"
+    log("Done!")
+    out_name = f"benchmarks/results/second-phase/latest/{generate_output_path(args)}"
```

Comment:
What is the "second phase"?

Reply:
I am not sure, I believe this was @Kriplingo s addition

Reply:
Since nobody knows what "second phase" refers to, @Kriplingo will revert this.